src/data-types/mailstream_ssl.c

```diff
@@ -950,17 +998,32 @@ static int wait_write(mailstream_low * s)
   write_enabled = (fds_read.fd_array[r - WAIT_OBJECT_0] == event);
 	WSAEventSelect(ssl_data->fd, event, 0);
 	CloseHandle(event);
+#elif USE_POLL
+  pfd[0].fd = ssl_data->fd;
+  pfd[0].events = POLLOUT;
+  pfd[0].revents = 0;
+
+  pfd[1].fd = fd;
+  pfd[1].events = POLLIN;
+  pfd[1].revents = 0;
+
+  r = poll(&pfd[0], 2, timeout.tv_sec * 1000 + timeout.tv_usec / 1000);
+  if (r < 0)
```

Comment:
I think you need to test (r <= 0) (similarly to select()) since you want to return an error if there's a timeout.